app/models/embed/purl.rb

```diff
@@ -137,6 +138,35 @@ def initialize(msg = 'The requested PURL resource was not embeddable.')
         super
       end
     end
+
+    class Duration < ISO8601::Duration
+      def to_s
+        errors = []
+        errors << 'Duration#to_s does not support specifying durations in weeks' if atoms[:weeks] != 0
+        errors << 'Duration#to_s does not support specifying negative durations' if atoms.values.any? { |val| val < 0 }
+        unless errors.empty?
+          errors.each { |e| Rails.logger.warn e }
+          return nil
+        end
```

Comment:
It feels like this method may have an extra responsibility that could be abstracted out.

What would you think about pulling this logic into private validation methods so we could do something along the lines of `return nil unless valid_duration?` at the top of `#to_s`?

Totally up to you.